Add doc for new `contains` jinja filter and test (#25890)
home-assistant/home-assistant.github.io: source/_docs/configuration/templating.markdown

{% raw %}

```text
Closest out of given entities:
    {{ ['group.children', states.device_tracker] | closest }}
Closest to a coordinate:
    {{ ['group.children', states.device_tracker] | closest(23.456, 23.456) }}
Closest to some entity:
    {{ ['group.children', states.device_tracker] | closest(states.zone.school) }}
```

{% endraw %}

### Contains

> Jinja provide by default a [`in` operator](https://jinja.palletsprojects.com/en/latest/templates/#other-operators) how return `True` when one element is `in` a provided list.
> The `contains` test and filter allow you to do the exact opposite and test for a list containing an element. This is particularly usefully in `select` or `selectattr` filter, as well as to check a if a device have a specific attribute, a `supported_color_modes`, a specific light effect...

Reviewer: ```suggestion
Jinja provides by default a [`in` operator](https://jinja.palletsprojects.com/en/latest/templates/#other-operators) how return `True` when one element is `in` a provided list.
The `contains` test and filter allow you to do the exact opposite and test for a list containing an element. This is particularly useful in `select` or `selectattr` filter, as well as to check if a device has a specific attribute, a `supported_color_modes`, a specific light effect.
```

Author: Fixed thank

Some examples:
{% raw %}

> - `{{ state_attr('light.dining_room', 'effect_list') | contains('rainbow') }}` will return `true` if the light have a `rainbow` effect

Reviewer: ```suggestion
- `{{ state_attr('light.dining_room', 'effect_list') | contains('rainbow') }}` will return `true` if the light has a `rainbow` effect.
```

Author: Fixed thank

- `{{ expand('light.office') | selectattr("attributes.supported_color_modes", 'contains', 'color_temp') | list }}` will return all light that support color_temp in the office group.
- ```text
    {% set current_month = now().month %}
    {% set extra_ambiance = [
      {'name':'Halloween', 'month': [10,11]},
      {'name':'Noel', 'month': [1,11,12]}
    ]%}
    {% set to_add = extra_ambiance | selectattr('month', 'contains', current_month ) | map(attribute='name') | list  %}
    {% set to_remove = extra_ambiance | map(attribute='name') | reject('in', to_add) | list %}
    {{ (state_attr('input_select.light_theme', 'options') + to_add ) | unique | reject('in', to_remove) | list }}
  ```

>   This more complex example use the `contains` filter to match the current month with a list. In this case it's used to generate a list of light theme to give to the `Input select: Set options` service.

Reviewer: ```suggestion
  This more complex example uses the `contains` filter to match the current month with a list. In this case, it's used to generate a list of light theme to give to the `Input select: Set options` service.
```

Author: Fixed thank
